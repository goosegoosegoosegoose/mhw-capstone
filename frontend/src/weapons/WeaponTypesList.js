import React, { useState, useEffect } from "react";
import MhwApi from "../api";
import SearchForm from "../common/SearchForm";
import WeaponTypeCard from "./WeaponTypeCard";

const WeaponTypeList = () => {
  const [weaponTypes, setWeaponTypes] = useState([]);

  useEffect(function fetchWeaponTypesWhenMounted() {
    async function fetchWeaponTypes() {
      const res = await MhwApi.request("weapons");
      setWeaponTypes(res)
    }
    fetchWeaponTypes()
  }, []);

  const getWeaponTypes = (res) => {
    setWeaponTypes([...res])
  }

  if (weaponTypes.length === 0) {
    return (
      <div className="container text-center">
        <h1>Weapon Types</h1>
        <SearchForm type="weapons" header="type" get={getWeaponTypes} />
        <p>Sorry, no results were found! Please re-submit to refresh.</p>
      </div>
    )
  }

  return (
    <div className="container text-center">
      <h1>Weapon Types</h1>
      <SearchForm type="weapons" header="type" get={getWeaponTypes} />
      <div className="row justify-content-center">
        {weaponTypes.map(t => 
          <WeaponTypeCard
            key={t.type}
            type={t.type}
            img={t.img}
          />
        )}
      </div>
  </div>
  )
}

export default WeaponTypeList;