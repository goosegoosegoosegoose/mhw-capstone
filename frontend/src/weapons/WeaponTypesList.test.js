import { render } from '@testing-library/react';
import WeaponTypesList from "./WeaponTypesList";

test('smoke test', () => {
  render(      
    <WeaponTypesList />
  );
});

test('snapshot', () => {
  const { asFragment } = render(<WeaponTypesList />);
  expect(asFragment()).toMatchSnapshot();
});