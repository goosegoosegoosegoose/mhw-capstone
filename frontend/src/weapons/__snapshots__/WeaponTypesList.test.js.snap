// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="container text-center"
  >
    <h1>
      Weapon Types
    </h1>
    <form>
      <div
        class="my-3 row"
      >
        <div
          class="input-group col-2"
        >
          <input
            class="form-control"
            placeholder="Enter search"
            value=""
          />
          <button
            class="mx-1 btn btn-dark btn-sm"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
    <p>
      Sorry, no results were found! Please re-submit to refresh.
    </p>
  </div>
</DocumentFragment>
`;
